swim/member.go

```diff
@@ -122,10 +123,14 @@ func (l LabelMap) checksum() int32 {
 	// make sure the buffer is empty after getting a buffer from our pool
 	lb.Reset()
 	for key, value := range l {
-		// decide if we want to escape both fields to enforce uniqueness of the
-		// key-value checksum.
+		// The bytes used for checksumming are the following:
+		// <4 bytes: length of key>
+		binary.Write(lb, binary.BigEndian, int32(len(key)))
```

Comment:
That was the exact behavior I was aiming for. This is implemented in nodejs via: https://github.com/uber/ringpop-common/commit/7e45a443939f651fffff0b7a1ec1a8e7bf8cb816 exactly as you describe by taking the length of the buffer.